### PR TITLE
Microphone stream selection demo

### PR DESCRIPTION
**Pull request by davidkline-ms**
This change adds a demo scene that senses the volume of the user's voice and brightens or dims the spatial mesh.

NOTE: This demo requires that the com.microsoft.mixedreality.toolkit.micstream package (https://github.com/microsoft/MixedRealityToolkit/pull/262) be installed. This package is only available for Unity 2019.4.

Fixes: #2505

### CODE REVIEW

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs [hunk 1]
```diff
@@ -0,0 +1,239 @@
...
+            micStream = null;
+
+#if UNITY_EDITOR
+            // Restore the initial material settings.
+            if (VisibleMaterial != null)
+            {
+                VisibleMaterial.SetColor("_WireColor", defaultMaterialColor);
+                VisibleMaterial.SetInt("_WireThickness", defaultWireThickness);
+            }
+#endif // UNITY_EDITOR
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:161]: Is this editor-only to ensure we don't accidentally save changes to the material file directly? Can we instead guard against that however we're accessing / editing the material?

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs [hunk 2]
```diff
@@ -0,0 +1,239 @@
...
+namespace Microsoft.MixedReality.Toolkit.Examples
+{
+    /// <summary>
+    /// Demonstration class using <see cref="WindowsMicrophoneStream"/> to select the voice microphone and
+    /// adjust the spatial awareness mesh based on the amplitude of the user's voice.
+    /// </summary>
+    [RequireComponent(typeof(AudioSource))]
+    public class MicrophoneAmplitudeDemo : MonoBehaviour
+    {
+#if MICSTREAMDLL_PRESENT
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:17]: super-nit, but can we remove `DLL` from the define? makes it a little more readable, and i don't _think_ we've used DLL in past ones (but i could be wrong!)

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:161]: correct. i will look into a non-editor way to do this

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs [hunk 3]
```diff
@@ -0,0 +1,239 @@
...
+
+using Microsoft.MixedReality.Toolkit.Audio;
+using Microsoft.MixedReality.Toolkit.SpatialAwareness;
+using UnityEngine;
+
+namespace Microsoft.MixedReality.Toolkit.Examples
+{
+    /// <summary>
+    /// Demonstration class using WindowsMicrophoneStream (from com.microsoft.mixedreality.toolkit.micstream) to select the 
+    // voice microphone and adjust the spatial awareness mesh based on the amplitude of the user's voice.
```
Comment (keveleigh) [inline Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:12]:
```suggestion
    /// voice microphone and adjust the spatial awareness mesh based on the amplitude of the user's voice.
```

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs (hunk 1 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs]: i removed the editor guards. the demo now restores the material values on destroy on device as well as in editor

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs [hunk 4]
```diff
@@ -0,0 +1,233 @@
...
+        }
+
+        private IMixedRealitySpatialAwarenessMeshObserver spatialMeshObserver = null;
+
+        /// <summary>
+        /// Instance of a spatial awareness mesh observer. Used to acquire the visible mesh material.
+        /// </summary>
+        private IMixedRealitySpatialAwarenessMeshObserver SpatialMeshObserver
+        {
+            get
```
Comment (CDiaz-MS) [inline Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:53]: Is SpatialMeshObserver meant to be public or private?  We usually capitalize public fields.

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs [hunk 5]
```diff
@@ -0,0 +1,233 @@
...
+        private Material visibleMaterial = null;
+
+        /// <summary>
+        /// The visible mesh material in use by the spatial mesh observer.
+        /// </summary>
+        private Material VisibleMaterial
+        {
+            get
+            {
+                if (visibleMaterial == null)
```
Comment (CDiaz-MS) [inline Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:72]: Is VisibleMaterial meant to be private or public?

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs (hunk 4 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:53]: good call! i'll clean this up. it is meant to be private

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs (hunk 5 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs:72]: same as the observer. thanks!

Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs]: removed the property in favor of a simple private field

File: Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs (hunk 4 above)
Comment (davidkline-ms) [reply Assets/MRTK/Examples/Demos/Audio/Scripts/MicrophoneAmplitudeDemo.cs]: removed the property in favor of a simple private field

**Review by CDiaz-MS — APPROVED** [review]

**Pull request merged**